File: output/gemma4-e2b/avatar/Python/Java/atcoder_ABC132_A.java
import java.util.Scanner;
import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;

public class atcoder_ABC132_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        if (!scanner.hasNextLine()) {
            return;
        }
        String S = scanner.nextLine();

        Map<Character, Integer> c = new HashMap<>();
        Set<Character> Sa = new HashSet<>();

        // Calculate frequencies and unique characters
        for (char ch : S.toCharArray()) {
            c.put(ch, c.getOrDefault(ch, 0) + 1);
            Sa.add(ch);
        }

        // Check condition 1: Number of unique characters must be 2
        if (Sa.size() != 2) {
            System.out.println("No");
            return;
        }

        // Check condition 2: All characters must appear exactly twice
        boolean allCountsAreTwo = true;
        for (char i : Sa) {
            if (c.get(i) != 2) {
                allCountsAreTwo = false;
                break;
            }
        }

        if (allCountsAreTwo) {
            System.out.println("Yes");
        } else {
            System.out.println("No");
        }
    }
}

File: output/gemma4-e2b/avatar/Java/Python/atcoder_ABC132_A.py
import sys
from collections import Counter

def solve():
    # Read the input string from standard input
    try:
        s = sys.stdin.read().strip()
    except:
        s = ""

    if not s:
        print("No")
        return

    # Count character frequencies
    counts = Counter(s)

    # Check if the number of distinct characters is exactly 2
    if len(counts) == 2:
        # Check if all counts are exactly 2
        if all(count == 2 for count in counts.values()):
            print("Yes")
        else:
            print("No")
    else:
        print("No")

solve()